third_party/amd/lib/TritonAMDGPUTransforms/StreamPipelineV2.cpp

```diff
@@ -662,7 +787,18 @@ bool StreamPipeliner::preprocessLoopAndBuildSchedule() {
   // Explicitly deallocate created allocations.
   for (auto alloc : sharedMemAllocs)
     builder.create<ttg::LocalDeallocOp>(forOp.getLoc(), alloc);
-  return true;
+
+  return success();
+}
+
+LogicalResult StreamPipeliner::pipelineLoop() {
+  if (failed(preprocessLoopAndBuildSchedule()))
+    return failure();
```

Comment:
It is possible to fail the pipelineExpander, which could have created a new for loop but it doesn't replace the old one until it is successful. So the result is a for loop that would be discarded by canonicalize since the outputs are not used. 

Would we want that to fail the compiler and not generate a binary? Since it is a global setting, it could also just be ignored. Maybe we fail the pass if the loop has explicit num_stages.

Reply:
okay that might be fine for now then.